### PR TITLE
feat: Add 422 as well understood for baloise

### PR DESCRIPTION
**Pull request by leubedane**
Add http status code 422 to well understood status codes. Allow 422 for all methods.

### CODE REVIEW

File: baloise.yml [hunk 1]
```diff
@@ -98,6 +99,56 @@ rules:
...
+          '403': [ ALL ]
+          '404': [ ALL ]
+          '405': [ ALL ]
+          '406': [ ALL ]
+          '408': [ ALL ]
+          '409': [ POST, PUT, DELETE, PATCH ]
+          '410': [ ALL ]
+          '412': [ PUT, DELETE, PATCH ]
+          '415': [ POST, PUT, DELETE, PATCH ]
+          '422': [ ALL ]
```
Comment (dweber019) [inline baloise.yml:138]: According of the definition https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 I would argue that this should only happen for POST, PUT, PATCH, as only there you should have content.

**Review by dweber019 — CHANGES_REQUESTED** [review]

File: baloise.yml (hunk 1 above)
Comment (joachimprinzbach) [reply baloise.yml:138]: what about query params with content?

Comment (dweber019) [reply baloise.yml:138]: We can see it this way but the official docs states that 415 (Unsupported Media Type) status code is inappropriate and 400 is too, as it's a semantical error, so one could think to narrow the verbs down to at most POST, PUT, DELETE, PATCH.

I think is comes down how we want to threat content-type related status codes regarding query parameters, as they can take content-types of text or json, with out declaring the content type by header.

Would we send a 415 if we provide a query parameter named ABC with some json as value but a string was expected?

Comment (joachimprinzbach) [reply baloise.yml:138]: It is not about the validity of the type / the structure, but rather about the content. 

Example: GET /locations?zipCode=413233

This is semantically correct, but from a business perspective 413233 is not a valid zipCode.
So in my understanding a 422 would be correct in this case. 

Other approach could be a 200 with an empty array as response.

Comment (dweber019) [reply baloise.yml:138]: For me this is fine, just wanted to propose to narrow it down, as we can make it open in the future anyway 😉 

In you example I would definitely return 200 with an empty array, as it seam to be a search/filter query and if noting is found I would return an empty array. If we do a validation regarding zipCodes, I would rather return a 400 validation error, as most frameworks handle such errors with a 400 (e.g. spring and hibernate-validator).

**Review by dweber019 — APPROVED** [review]

**Pull request merged**